### ISSUE TITLE
Invalid service account: default while deploying functions with firebase-tools@11.2.0

### ISSUE THREAD

**Issue by ibrahimduran** [firebase/firebase-tools#4697]
### [REQUIRED] Environment info

**firebase-tools:**
11.2.0

**Platform:**
Windows 11 - 21H2

### [REQUIRED] Test case

Firebase cli fails to deploy functions for new project. Downgrading to 11.1.0 works.

### [REQUIRED] Steps to reproduce

1. Install `firebase-tools@11.2.0`
2. Create firebase project or use existing one (?)
3. Initialize functions codebase using `firebase init`
4. Run deployment by `firebase deploy --debug --only functions`

### [REQUIRED] Expected behavior

Successful deployment. 

### [REQUIRED] Actual behavior

Fails deployment by `Invalid function service account requested: default.` error.

```
[2022-07-01T07:32:03.535Z] >>> [apiv2][body] POST https://cloudfunctions.googleapis.com/v1/projects/my-project/locations/us-central1/functions {"name":"projects/my-project/locations/us-central1/functions/helloWorld","sourceUploadUrl":"...","entryPoint":"helloWorld","runtime":"nodejs16","dockerRegistry":"ARTIFACT_REGISTRY","labels":{"deployment-tool":"cli-firebase"},"httpsTrigger":{"securityLevel":"SECURE_ALWAYS"},"serviceAccountEmail":"default","environmentVariables":{"FIREBASE_CONFIG":"{\"projectId\":\"my-project\",\"storageBucket\":\"my-project.appspot.com\",\"locationId\":\"europe-west\"}","GCLOUD_PROJECT":"my-project"},"timeout":"60s"}
[2022-07-01T07:32:03.816Z] <<< [apiv2][status] POST https://cloudfunctions.googleapis.com/v1/projects/my-project/locations/us-central1/functions 400
[2022-07-01T07:32:03.816Z] <<< [apiv2][body] POST https://cloudfunctions.googleapis.com/v1/projects/my-project/locations/us-central1/functions {"error":{"code":400,"message":"Invalid function service account requested: default. Please visit https://cloud.google.com/functions/docs/troubleshooting for in-depth troubleshooting documentation.","status":"INVALID_ARGUMENT"}}
!  functions: failed to create function projects/my-project/locations/us-central1/functions/helloWorld
Failed to create function projects/my-project/locations/us-central1/functions/helloWorld
```

**Comment by Schmale97** [comment]
I am also experiencing this when deploying from ubuntu based docker image

**Comment by KClough** [comment]
I'm seeing the same issue with `firebase-functions@3.22.0`

**Comment by mhotan** [comment]
I had to revert to 11.1.0 in order to deploy my functions successfully. The `--debug` logs are useful as it indicated there were new permission requirements likely introduced. I tried a couple permission updates unsuccessfully and it still looks like 11.2.0 is not respecting the GOOGLE_APPLICATION_CREDENTIALS environment variable.

Specifically, I updated my CI script to `npm install -g firebase-tools@11.1.0` in order to unblock my production deploys.

**Comment by tubbo** [comment]
We also noticed that in our CI, `firebase-tools` shows a warning when a function fails to deploy, but does not return a non-zero exit code. That prevented us from immediately the problem described in this issue. I don't see a way to make `firebase-tools` behave in such a manner...

```
Usage: firebase deploy [options]

deploy code and assets to your Firebase project

Options:
  -f, --force              delete Cloud Functions missing from the current working directory without confirmation
  -p, --public <path>      override the Hosting public directory specified in firebase.json
  -m, --message <message>  an optional message describing this deploy
  --only <targets>         only deploy to specified, comma-separated targets (e.g. "hosting,storage"). For functions, can specify filters with colons to scope function deploys to only those
                           functions (e.g. "--only functions:func1,functions:func2"). When filtering based on export groups (the exported module object keys), use dots to specify group names
                           (e.g. "--only functions:group1.subgroup1,functions:group2)"
  --except <targets>       deploy to all targets except specified (e.g. "database")
  -h, --help               output usage information
```

It would be pretty awesome if the Firebase CLI would exit 0 in these circumstances, or at least give us a `--ci` option to run for automated deployments.

**Comment by andreasalptun** [comment]
@mhotan Same problem here with 11.2.0, but downgrading worked for me. Thanks!

**Comment by akauppi** [comment]
@tubbo I do not quite understand your text above. 

>shows a warning when a function fails to deploy, but does not return a non-zero exit code.

I'd definitely want *anything* failing to fail the whole `firebase deploy`, with a non-zero exit code.

>pretty awesome if the Firebase CLI would exit 0 in these circumstances

If I don't get these straight, maybe others will also struggle. Is it just me?

**Comment by taeold** [comment]
Re-opening - will mark closed after we have a release shipped.

**Comment by Schmale97** [comment]
Is this meant to be closed? I can't see a new release

**Comment by taeold** [comment]
Fix released in 11.2.1
